runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java

```diff
@@ -109,6 +110,17 @@
 
   void setTemplateLocation(String value);
 
+  /**
+   * Service options are set by the user and configure the service. This decouples service side
+   * feature availability from the Apache Beam release cycle.
+   */
+  @Description(
```

Comment:
Are we going to have detailed documentation on what could be the service options here? And will this option be overlapped partially with current existing options?

Reply:
Yes. The documentation will be on cloud.google.com for Dataflow. One of the pages for specifying execution parameters is being refactored, once that is complete we'll have a better idea of how to incorporate these service options.

There will be some overlap with `experiments` for the time being until the experiments that are 'GA' have first class pipeline options introduced.